.github/workflows/ci-sp1.yml

```
@@ -0,0 +1,19 @@
+name: CI - SP1
+
+on:
+  workflow_call:
+  push:
+    branches: ["main"]
+    paths:
+      - "provers/sp1/**"
+  pull_request:
+    paths:
+      - "provers/sp1/**"
+
+jobs:
+    build-test-sgx:
+      name: Build and test sp1
+      uses: ./.github/workflows/ci-build-test-reusable.yml
+      with:
+        version_name: "sp1"
+        version_toolchain: "nightly-2024-04-17"
```

```suggestion
        version_toolchain: "nightly-2024-04-18"
```
I just PR to update the `install.sh` to 04-18 cuz i remember some package requires 04-18 at some point so let's all use 04-18 just to be safe